presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java

```diff
@@ -634,7 +647,9 @@ public void close()
         final AtomicLong peakTotalMemoryReservation = new AtomicLong();
 
         @Nullable
-        WorkProcessorOperator operator;
+        volatile WorkProcessorOperator operator;
+        @Nullable
+        volatile OperatorInfo finalOperatorInfo;
```

Comment:
Stats are fetched relatively rarely (e.g. every 1 sec or so). I didn't want to synchronized on every page produced by operator (in case pages are tiny)